src/zenml/integrations/daytona/orchestrators/daytona_orchestrator.py

```diff
@@ -0,0 +1,300 @@
+            str(self.id),
+        )
+
+    def prepare_or_run_pipeline(
+        self,
+        deployment: "PipelineDeploymentResponse",
+        stack: "Stack",
+        environment: Dict[str, str],
+    ) -> None:
+        """Prepare and run the pipeline using Daytona SDK.
+
+        Args:
+            deployment: The pipeline deployment to prepare or run.
+            stack: The stack the pipeline will run on.
+            environment: Environment variables to set in the orchestration
+                environment.
+
+        Raises:
+            ValueError: If the Daytona API key is not set in the settings.
+        """
+        settings = cast(
+            DaytonaOrchestratorSettings, self.get_settings(deployment)
+        )
+        if not settings.api_key:
+            raise ValueError(
+                "Daytona orchestrator requires `api_key` to be set in the settings."
+            )
+
+        # Initialize Daytona client
+        daytona_config = DaytonaConfig(
+            api_key=settings.api_key or "",
+            server_url=settings.server_url or "https://daytona.work/api",
+            target=settings.target or "us",
+        )
+        daytona = Daytona(daytona_config)
+
+        # Create a workspace for the pipeline
+        orchestrator_run_name = get_orchestrator_run_name(
+            pipeline_name=deployment.pipeline_configuration.name
+        )
+
+        logger.info("Creating Daytona workspace...")
+        workspace_params = CreateWorkspaceParams(
+            language="python",
+            id=orchestrator_run_name,
+            name=orchestrator_run_name,
+            image=settings.image,
+            os_user=settings.os_user,
+            env_vars=settings.env_vars,
+            labels=settings.labels,
+            public=settings.public,
+            target=settings.target or "us",
+            timeout=settings.timeout,
+            auto_stop_interval=settings.auto_stop_interval,
+        )
+
+        # Add resource configuration if any resource settings are specified
+        if any(
+            x is not None
+            for x in [
+                settings.cpu,
+                settings.memory,
+                settings.disk,
+                settings.gpu,
+            ]
+        ):
+            workspace_params.resources = WorkspaceResources(
+                cpu=settings.cpu,
+                memory=settings.memory,
+                disk=settings.disk,
+                gpu=settings.gpu,
+            )
+
+        workspace = daytona.create(workspace_params)
+
+        # Create a wheel for the package in a temporary directory
+        with tempfile.TemporaryDirectory() as temp_dir:
+            wheel_path = self.create_wheel(temp_dir=temp_dir)
+
+        # Set up environment variables
+        env_vars = environment.copy()
+        env_vars[ENV_ZENML_DAYTONA_RUN_ID] = orchestrator_run_name
+        env_vars[ENV_ZENML_CUSTOM_SOURCE_ROOT] = (
+            DAYTONA_ZENML_DEFAULT_CUSTOM_REPOSITORY_PATH
```

Comment:
I'm not sure how this is supposed to work, I don't think anything exists at that path?